pkg/operator/controller/callbacks.go

```diff
@@ -355,3 +362,115 @@ func restoreOlderVersions(currentCrd, desiredCrd *extv1.CustomResourceDefinition
 	}
 	return desiredCrd
 }
+
+func reconcilePvcMutatingWebhook(args *callbacks.ReconcileCallbackArgs) error {
+	if args.State != callbacks.ReconcileStatePostRead {
+		return nil
+	}
+
+	deployment, ok := args.DesiredObject.(*appsv1.Deployment)
+	if !ok || deployment.Name != common.CDIApiServerResourceName {
+		return nil
+	}
+
+	enabled, err := featuregates.IsWebhookPvcRenderingEnabled(args.Client)
+	if err != nil {
+		return cc.IgnoreNotFound(err)
+	}
+
+	whc := &admissionregistrationv1.MutatingWebhookConfiguration{}
+	key := client.ObjectKey{Name: "cdi-api-pvc-mutate"}
+	err = args.Client.Get(context.TODO(), key, whc)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	exists := err == nil
+	if !enabled {
+		if !exists {
+			return nil
+		}
+		err = args.Client.Delete(context.TODO(), whc)
+		return client.IgnoreNotFound(err)
+	}
+
+	if !exists {
+		if err := initPvcMutatingWebhook(whc, args); err != nil {
+			return err
+		}
+		return args.Client.Create(context.TODO(), whc)
+	}
+
+	whcCopy := whc.DeepCopy()
+	if err := initPvcMutatingWebhook(whc, args); err != nil {
+		return err
+	}
+	if !reflect.DeepEqual(whc, whcCopy) {
+		return args.Client.Update(context.TODO(), whc)
```

Comment:
I was curious how frequently update would get called so I pulled your branch added a log here and ran the code.  Turns out it was being called every reconcile even if nothing changed.  Turned out it was because you are setting:

```golang
whc.APIVersion = "admissionregistration.k8s.io/v1"
whc.Kind = "MutatingWebhookConfiguration"
```

Controller runtime strips that out.  So I'd suggest leaving it out here too.

FYI: it is added in `operator/resources` because the objects may be serialized to yaml

Reply:
Great catch & great lesson learned. Fixed.